stack_configuration.go

```diff
@@ -155,3 +161,95 @@ func (s stackConfigurations) List(ctx context.Context, stackID string, options *
 
 	return result, nil
 }
+
+type CreateStackConfigurationOptions struct {
+	SelectedDeployments []string `jsonapi:"attr,selected-deployments,omitempty"`
+	SpeculativeEnabled  *bool    `jsonapi:"attr,speculative,omitempty"`
+}
+
+// CreateAndUpload packages and uploads the specified Terraform Stacks
+// configuration files in association with a Stack.
+func (s stackConfigurations) CreateAndUpload(ctx context.Context, stackID, path string, opts *CreateStackConfigurationOptions) (*StackConfiguration, error) {
+	if opts == nil {
+		opts = &CreateStackConfigurationOptions{}
+	}
+	u := fmt.Sprintf("stacks/%s/stack-configurations", url.PathEscape(stackID))
+	req, err := s.client.NewRequest("POST", u, opts)
+	if err != nil {
+		return nil, fmt.Errorf("error creating stack configuration request for stack %q: %w", stackID, err)
+	}
+
+	sc := &StackConfiguration{}
+	err = req.Do(ctx, sc)
+	if err != nil {
+		return nil, fmt.Errorf("error creating stack configuration for stack %q: %w", stackID, err)
+	}
+
+	uploadURL, err := s.pollForUploadURL(ctx, sc.ID)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving upload URL for stack configuration %q: %w", sc.ID, err)
+	}
+
+	body, err := packContents(path)
+	if err != nil {
+		return nil, err
+	}
+
+	err = s.UploadTarGzip(ctx, uploadURL, body)
```

Comment:
Do you want this to be a part of the public interface? Because it's public but not an interface method